src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -184,7 +184,43 @@ public bool ValidateInstructions(ReadOnlySpan<byte> container, out EofHeader? he
         }
 
         if (ExtractHeader(container, spec, out header))
+        {
+            if (!spec.IsEip3670Enabled)
+            {
+                return true;
+            }
+
+            var (startOffset, sectionSize) = (header.Value.CodeSection.Start, header.Value.CodeSection.Size);
+            ReadOnlySpan<byte> code = container.Slice(startOffset, sectionSize);
+            Instruction? opcode = null;
+            for (int i = 0; i < sectionSize;)
+            {
+                opcode = (Instruction)code[i];
+                i++;
+                // validate opcode
+                if (!opcode.Value.IsValid(spec))
+                {
+                    if (LoggingEnabled)
+                    {
+                        _logger.Trace($"EIP-3670 : CodeSection contains undefined opcode {opcode}");
+                    }
+                    header = null; return false;
+                }
+
+                if (opcode is >= Instruction.PUSH1 and <= Instruction.PUSH32)
+                {
+                    int len = code[i - 1] - (int)Instruction.PUSH1 + 1;
+                    i += len;
+                }
+
+                if (i > sectionSize)
+                {
+                    return false;
```

Comment:
Set header to null?

Reply:
Header is set to null in `IsValidEof` method.